### DISCUSSION TITLE
Context.channel is not always a TextChannel?

### DISCUSSION THREAD

**Discussion by BIRD-UP73** [Rapptz/discord.py#6504] [Q&A]
The [documentation](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=context#discord.ext.commands.Context.channel) for `Context.channel` shows this property is a `TextChannel`.

However, could this not also be a `DMChannel`?

The typing for [Message.channel](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=context#discord.ext.commands.Context.channel) is `Union[abc.Messageable]`

**Comment by esmaycat** [comment]
As the method `Context.channel` is shorthand for `Message.channel` and the [documentation](https://discordpy.readthedocs.io/en/latest/api.html#discord.Message.channel) for `Message.channel` states that it 

> [The TextChannel that the message was sent from.] Could be a `DMChannel` or `GroupChannel` if it’s a private message.

Seeing as discord bots cannot be in a `GroupChannel` then yes, the `Context.channel` could also be a `DMChannel` because of the method that it is shorthand for: `Message.channel`'s documentation states that it can be one of the options states above (excluding the `GroupChannel` for a bot account).

Also the class `Context` implements the `Messageable` and that in turn implements the following:

![Screenshot 2021-03-03 at 23 31 11](https://user-images.githubusercontent.com/71032999/109886795-8c7f3a00-7c78-11eb-8a7a-51472d1e9168.png) 

including the `DMChannel`

**Reply by BIRD-UP73** [reply]
So should the doc be changed?

If you try to use `ctx.channel.name` for example, you get an error in the case of a `DMChannel`.

**Reply by esmaycat** [reply]
You could try `ctx.message.channel` to get the a `Channel` object from the message. ¯\_(ツ)_/¯
